src/main/java/org/haffson/adventofcode/days/day02/Day02.java

```diff
@@ -0,0 +1,177 @@
+            return "Part 2 answer: " + getNumberPwd2();
+        } catch (FileNotFoundException e) {
+            return "Error in method 2";
+        }
+    }
+
+
+    /** Method to read raw data from file into list
+     *
+     * @return raw data as list
+     */
+    public List<String> getRawDataAsList(String path) {
+        List<String> rawData = new ArrayList<>();
+
+        try {
+            Scanner s = new Scanner(new File(path)).useDelimiter("\n");
+            while (s.hasNext()) {
+                rawData.add(s.next());
+            }
+            s.close();
+        } catch (FileNotFoundException e) {
+            System.out.println("File not found!");
+        }
+        return rawData;
+    }
+
+    /**
+     * Method getData() to get min and max number, letter and password from raw data
+     * First, create class Data to avoid having multiple return values
+     */
+    public static class Data {
+        int min;
+        int max;
+        String letter;
+        String pwd;
+
+        public Data(int min, int max, String letter, String pwd) {
+            this.min = min;
+            this.max = max;
+            this.letter = letter;
+            this.pwd = pwd;
+        }
+    }
+
+    public List<Data> getData(List<String> rawData) {
+
+        List<Data> dataArrayList = new ArrayList<>();
+
+        for (int i = 0; i < rawData.size(); i++) {
+
+            String line = rawData.get(i);
+
+            // match patterns to determine min and max number of the letter in pwd, the searched letter and pwd itself
+            String pattern = "(\\d+)-(\\d+)\\s+(\\w):\\s+(\\w*)";
+            int min = Integer.parseInt(line.replaceAll(pattern, "$1"));
+            int max = Integer.parseInt(line.replaceAll(pattern, "$2"));
+            String letter = line.replaceAll(pattern, "$3");
+            String pwd = line.replaceAll(pattern, "$4");
+
+            dataArrayList.add(new Data(min, max, letter, pwd));
+        }
+        return dataArrayList;
+    }
+
+
+    /**
+     * Primary method for Day 1, Part 1.
+     * Gets the number of correct passwords from a list
+     *
+     * @return the number of correct pwds in list
+     */
+
+    private int getNumberPwd1() throws FileNotFoundException {
+        // get data
+        List<Data> data = getData(rawData);
+
+        int countCorrPwd = 0; // answer to puzzle day 2.1: number of correct passwords in list
+
+        // loop through passwords
+        for (int i=0; i < data.size(); i++ ) {
+            int count = 0;  // number of letter appearance in pwd
+
+            // count how often specific letter appears in pwd
+            for (int j = 0; j < data.get(i).pwd.length(); j++) {
+                if (data.get(i).pwd.charAt(j) == data.get(i).letter.charAt(0)) count++;
+            }
+             // check if count meets criteria for correct pwd
+            if (count >= data.get(i).min && count <= data.get(i).max) {
+                countCorrPwd++;
+            }
+        }
```

Comment:
This can also be simplified by using the enhanced for loop (see IntelliJ's suggestion, it should be highlighted in yellow):
```suggestion
        // loop through passwords
        for (Data datum : data) {
            int count = 0;  // number of letter appearance in pwd

            // count how often specific letter appears in pwd
            for (int j = 0; j < datum.pwd.length(); j++) {
                if (datum.pwd.charAt(j) == datum.letter.charAt(0)) count++;
            }
            // check if count meets criteria for correct pwd
            if (count >= datum.min && count <= datum.max) {
                countCorrPwd++;
            }
        }
```